security/fma-kona-asterisc.md

```diff
@@ -0,0 +1,213 @@
+
+
+### Kona Execution Diverges from op-program
+
+- **Description:**
+  The Kona Fault Proof Program executes blocks in a stateless L2 block executor that uses optimism-revm as a backend.
+  If optimism-revm or the L2 block executor diverge in behavior from the `op-program`, the `kona` + `asterisc` proof system will be invalid.
+
+- **Risk Assessment:**
+  Low likelihood, low impact.
+  A small divergence in execution behavior can cause invalid game results.
+  But since kona + asterisc is in the coldpath, it doesn't have impact on anything in the production system.
+
+- **Mitigations:**
+  State testing against op-revm.
+  Action tests covering the L2 block executor are run on every pull request in the [kona] repository.
+  `op-reth` which uses the same op-execution environment as `kona`.
+
+- **Detection:**
+  Running `op-reth` so it's synced and following tip would provide a way to watch for chain splits.
+  A chain split occuring means `kona`'s execution (the same as `op-reth`) diverges from the `op-program`.
+
+- **Recovery Path(s):**
+  If `kona` + `asterisc` is in the coldpath, it would need to be fixed and re-deployed.
+  If `op-program` + `cannon` cannot be placed in the hotpath, and `kona` + `asterisc` fails, the permissioned fallback must be engaged.
+  [The "Kona and Asterisc Hot Path Failure"](#-Kona-and-Asterisc-Hot-Path-Failure) above covers the failure mode for `kona` + `asterisc` when placed in the hot path.
+
+
+### Breaking Backwards Compatibility for the op-challenger
+
+- **Description:**
+  The `op-challenger` has been updated to support the Kona + Asterisc Fault Dispute Game type ([Game Type 3]).
+
+- **Risk Assessment:**
+  Low likelihood as the action tests that are run in CI very frequently run the `op-challenger` with the `kona` + `asterisc` fault dispute game.
+  Medium impact since the off-chain `op-challenger` service can be updated with enough buffer time due to the large game time and nature of the chess clock.
+
+- **Mitigations:**
+  The `op-challenger` is architected in such a way as to isolate game types and moreso the inididual game players.
+  To allow the challenger to scale and play many different game types at once, a list of game types to play is passed into the `op-challenger` as a CLI flag.
+  When games are detected on-chain, the `op-challenger` will create a "game player" for this game type.
+  Players are run in individual threads so a breaking `kona` + `asterisc` game player in the challenger will not cause a liveliness issue for the `op-challenger` to play other game types.
+  Another key mitigation is the VM runner. This is an offline simulation that runs asterisc + kona with the op-challenger. It's been running since around August 2024.
+
+- **Detection:**
+  The Dispute Monitor ([op-dispute-mon] service provides an isolated monitoring service that will detect if a game is not being played by the honest `op-challenger`.
```

Comment:
> a game is not being played by the honest op-challenger

What does that means exactly? I thought that the `op-challenger` will challenge the invalid game by attacking them as the `CHALLENGER`. 
If this is true what do you mean by **played by the honest `op-challenger`** sounds to me that only the game that are played and created by the `op-challenger` would be considered as honest. 
But this is not possible except if this similar to a PermissionedGame and only the `challenger` can create and play the game?

Reply:
Same as the `op-program` + `cannon` dispute game, if the honest actor does not participate, the game can be finalized incorrectly. The fault proof system relies on the presence of the honest actor.

> PS: Do we also plan to run a canary to make sure the liveness of the dispute-mon is correct with Asterisc FP program?

Yes. We will do the same thing that we do for the existing game. Also, we run the challenger, kona, and asterisc in the [VM Runner](https://optimistic.grafana.net/d/bdtk5xud0cg00e/vm-runner?from=now-2d&to=now&timezone=browser&var-namespace=$__all), which continuously integrates the components.